qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Factory and credentials manager for IBM Q Experience."""
+
+import logging
+from collections import OrderedDict
+
+from .accountprovider import AccountProvider
+from .api_v2.clients import AuthClient, VersionClient
+from .credentials import Credentials
+from .exceptions import IBMQAccountError, IBMQApiUrlError, IBMQProviderError
+from .ibmqprovider import IBMQProvider
+
+logger = logging.getLogger(__name__)
+
+QX_AUTH_URL = 'https://auth.quantum-computing.ibm.com/api'
+
+
+class IBMQFactory:
+    """Factory and credentials manager for IBM Q Experience."""
+
+    def __init__(self):
+        self._credentials = None
+        self._providers = OrderedDict()
+        self._v1_provider = IBMQProvider()
+
+    # Account management functions.
+
+    def enable_account(self, token, url=QX_AUTH_URL, **kwargs):
+        """Authenticate against IBM Q Experience for use during this session.
+
+        Args:
+            token (str): IBM Q Experience API token.
+            url (str): URL for the IBM Q Experience auth server.
+            **kwargs (dict): additional settings for the connection:
+                * proxies (dict): proxy configuration.
+                * verify (bool): verify the server's TLS certificate.
+
+        Returns:
+            AccountProvider: the provider for the default open access project.
+
+        Raises:
+            IBMQAccountError: if an IBM Q Experience 2 account is already in
+                use, or if attempting using both classic API and new API
+                accounts.
+            IBMQApiUrlError: if the credentials are from IBM Q Experience 2,
+                but do not belong to the authentication URL.
+        """
+        # Check if an IBM Q Experience 2 account is already in use.
+        if self._credentials:
+            raise IBMQAccountError('An IBM Q Experience 2 account is already '
```

Comment:
Not really - it was an attempt to be explicit about requiring "the new api". Since there will probably be some renaming back and forth overall, I'll open a follow-up to ensure we are consistent with the wording in the warnings and keep track.